mir-module/src/main/resources/xsl/mods2marcxml.xsl

```diff
@@ -0,0 +1,2820 @@
+		</xsl:call-template>
+	</xsl:template>
+
+	<xsl:template name="makeSize">
+		<xsl:param name="string"/>
+		<xsl:param name="length"/>
+		<xsl:variable name="nstring" select="normalize-space($string)"/>
+		<xsl:variable name="nstringlength" select="string-length($nstring)"/>
+		<xsl:choose>
+			<xsl:when test="$nstringlength&gt;$length">
+				<xsl:value-of select="substring($nstring,1,$length)"/>
+			</xsl:when>
+			<xsl:when test="$nstringlength&lt;$length">
+				<xsl:value-of select="$nstring"/>
+				<xsl:call-template name="buildSpaces">
+					<xsl:with-param name="spaces" select="$length - $nstringlength"/>
+					<xsl:with-param name="char">|</xsl:with-param>
+				</xsl:call-template>
+			</xsl:when>
+			<xsl:otherwise>
+				<xsl:value-of select="$nstring"/>
+			</xsl:otherwise>
+		</xsl:choose>
+	</xsl:template>
+
+	<xsl:template match="mods:mods">
+		<marc:record>
+			<marc:leader>
+				<!-- 00-04 -->
+				<xsl:text>     </xsl:text>
+				<!-- 05 -->
+				<xsl:text>n</xsl:text>
+				<!-- 06 -->
+				<xsl:apply-templates mode="leader" select="mods:typeOfResource[1]"/>
+				<!-- 07 -->
+				<xsl:choose>
+					<xsl:when test="mods:originInfo/mods:issuance='monographic'">m</xsl:when>
+					<xsl:when test="mods:originInfo/mods:issuance='continuing'">s</xsl:when>
+					<xsl:when test="mods:typeOfResource/@collection='yes'">c</xsl:when>
+					<xsl:when test="mods:originInfo/mods:issuance='multipart monograph'">m</xsl:when>
+					<xsl:when test="mods:originInfo/mods:issuance='single unit'">m</xsl:when>
+					<xsl:when test="mods:originInfo/mods:issuance='integrating resource'">i</xsl:when>
+					<xsl:when test="mods:originInfo/mods:issuance='serial'">s</xsl:when>
+					<xsl:otherwise>m</xsl:otherwise>
+				</xsl:choose>
+				<!-- 08 -->
+				<xsl:text> </xsl:text>
+				<!-- 09 -->
+				<xsl:text> </xsl:text>
+				<!-- 10 -->
+				<xsl:text>2</xsl:text>
+				<!-- 11 -->
+				<xsl:text>2</xsl:text>
+				<!-- 12-16 -->
+				<xsl:text>     </xsl:text>
+				<!-- 17 -->
+				<xsl:text>u</xsl:text>
+				<!-- 18 -->
+				<xsl:text>u</xsl:text>
+				<!-- 19 -->
+				<xsl:text> </xsl:text>
+				<!-- 20-23 -->
+				<xsl:text>4500</xsl:text>
+			</marc:leader>
+			<xsl:call-template name="controlRecordInfo"/>
+			<xsl:if test="mods:genre[@authority='marc']='atlas'">
+				<marc:controlfield tag="007">ad||||||</marc:controlfield>
+			</xsl:if>
+			<xsl:if test="mods:genre[@authority='marc']='model'">
+				<marc:controlfield tag="007">aq||||||</marc:controlfield>
+			</xsl:if>
+			<xsl:if test="mods:genre[@authority='marc']='remote sensing image'">
+				<marc:controlfield tag="007">ar||||||</marc:controlfield>
+			</xsl:if>
+			<xsl:if test="mods:genre[@authority='marc']='map'">
+				<marc:controlfield tag="007">aj||||||</marc:controlfield>
+			</xsl:if>
+			<xsl:if test="mods:genre[@authority='marc']='globe'">
+				<marc:controlfield tag="007">d|||||</marc:controlfield>
+			</xsl:if>
+			<marc:controlfield tag="008">
```

Comment:
controlfields are typical generated from bib system. perhaps we should delete them?